src/commonTest/kotlin/arrow/exact/ExactBuilderDslSpec.kt

```diff
@@ -0,0 +1,66 @@
+package arrow.exact
+
+import arrow.core.Either
+import arrow.core.raise.ensureNotNull
+import io.kotest.assertions.arrow.core.shouldBeLeft
+import io.kotest.assertions.failure
+import io.kotest.core.spec.style.FreeSpec
+import io.kotest.data.row
+import io.kotest.datatest.withData
+import io.kotest.matchers.shouldBe
+import kotlin.jvm.JvmInline
+
+sealed interface Artificial {
+  object Admin : Artificial
+  data class User(val id: Int) : Artificial
+}
+
+@JvmInline
+value class ArtificialConstraintType private constructor(val value: Artificial) {
+  companion object : Exact<String, ArtificialConstraintType> by exactBuilder({ builder ->
+    builder.mustBe(String::isNotBlank)
+      .transform(String::trim)
+      .transformOrRaise {
+        when {
+          it.startsWith("a/") -> Artificial.Admin
+          it.startsWith("u/") -> {
+            val userId = it.drop(2).toIntOrNull()
+            ensureNotNull(userId) { ExactError("Invalid user id in: $it") }
```

Comment:
On this example it becomes even more obvious that we need an ability to have custom errors like

``` kotlin
ArtificialConstraintType.from(" ").mapLeft {
  when(it) {
    is InvalidUser -> TODO()
    is BadInput -> TODO()  
  }
}
```
We will face with two problems

1. How to not overcomplicate `Exact` signature. Probably we can introduce another interface that will hold a type for a custom error like `class MyType : ExactAware<MyCustromError>`
2. How to chain errors when combining multiple `Exact`-s